app/components/works/agreement_component.rb

```diff
@@ -20,7 +16,7 @@ def work
     end
 
     delegate :depositor, to: :work
-    delegate :last_work_terms_agreement, to: :depositor
+    delegate :agreed_to_terms_recently?, :last_work_terms_agreement, to: :depositor
 
```

Comment:
this goes straight to the underlying method on the user model, which is what we want anyway

Reply:
Were you able to prove that this change, with all the conditions to trigger the exception previous, keeps that exception from being triggered?

Or are you expecting the not-yet-written test to do it?

It would be great if you wanted to write the test as part of the PR ...

Reply:
I was able to prove this change didn't blow up in browser.  I would suggest we leave it uncoupled from the ticket for the added test (though I am happy to have a look at that next).

Reply:
Or not...i guess this fix isn't an urgent bug fix, so either way is fine.